sql/core/src/main/scala/org/apache/spark/sql/jdbc/DerbyDialect.scala

```diff
@@ -22,11 +22,22 @@ import java.util.Locale
 
 import org.apache.spark.sql.connector.catalog.Identifier
 import org.apache.spark.sql.errors.{QueryCompilationErrors, QueryExecutionErrors}
+import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 
 
 private case class DerbyDialect() extends JdbcDialect with NoLegacyJDBCError {
 
+  /**
+   * Returns jdbc version that this dialect uses.
+   */
+  override def jdbcVersion(): Int = SQLConf.get.getConf(SQLConf.DERBY_JDBC_VERSION)
```

Comment:
Do we need jdbcVersion per dialect?

Reply:
There are updates on Jdbc common level.
How would we version connector if this is not per dialect ?

Blast radius of these changes would be much higher.

Reply:
Yeah, I agree, one config for all connectors for common code would be worse solution.
For example, customer can upgrade MySQL independently from Postgres, even common parts of code.